### PR TITLE
CONSOLE-4139: replace react-copy-to-clipboard with PatternFly Clipboa…

### PR DESCRIPTION
**Pull request by rhamilto**
…rdCopyButton component

There are additional instances of `react-copy-to-clipboard` that will be addressed by:
* https://issues.redhat.com/browse/ODC-7698
* https://issues.redhat.com/browse/CNV-49317

After

https://github.com/user-attachments/assets/44c8f67c-8f38-4ab6-ba06-353865deb7b8

https://github.com/user-attachments/assets/1b5393cc-81f6-4048-b50c-f3a8d2fc761c

### CODE REVIEW

**Comment by rhamilto** [comment]
/assign @TheRealJon 
/assign @yapei 

Adding approvals since this is tech debt:
/label px-approved
/label docs-approved

**Comment by yapei** [comment]
tested the PR and no regression issues found
/label qe-approved

**Comment by yapei** [comment]
I see inconsistent behavior on secrets and routes page
for an external link, after content is copied, it shows 'copied to clipboard' while for secret data it shows 'copied'

https://github.com/user-attachments/assets/35952f57-7ae9-46bc-9b54-246f99df4bf5

**Comment by rhamilto** [comment]
> I'm testing on a cluster-bot cluster `launch openshift/console#14327` and seeing different behavior for an external link, after content is copied, it shows 'copied to clipboard' (for secret data it shows 'copied')

Good catch, @yapei.  It turns out the `ExternalLinkWithCopy` component has been duplicated in https://github.com/openshift/networking-console-plugin and [the bug exists there](https://github.com/openshift/networking-console-plugin/blob/af17a520da3c58fc6f44c3aaf5215100573539a9/src/utils/components/ExternalLinkWithCopy/ExternalLinkWithCopy.tsx#L25) (in addition to that duplicated component needing to be updated to use `CopyClipboardButton`).  I've opened https://issues.redhat.com/browse/CNV-49317 to address.

**Comment by rhamilto** [comment]
/hold

I should be able to convert the exiting unit test to RTL using https://testing-library.com/docs/user-event/clipboard/

File: frontend/__tests__/components/utils/copy-to-clipboard.spec.tsx
[the remark this answers is not in the document]
Comment (rhamilto) [reply frontend/__tests__/components/utils/copy-to-clipboard.spec.tsx:1]: @jhadvig, mind reviewing?

[the remark this answers is not in the document]
Comment (jhadvig) [reply frontend/__tests__/components/utils/copy-to-clipboard.spec.tsx:1]: I agree with removing the test, since PF should be covering those 👍

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**